media.graphql

```diff
@@ -0,0 +1,185 @@
+# IPI  - Interested Parties Information
+# DPID - DDEX Party Identifier
+# ISRC - International Standard Recording Code
+# ISWC - International Standard Musical Work Code
+# GRid - Global Release Identifier
+# UPC  - Universal Product Code
+#
+# Other entity properties (like performer name or recording title) are linked
+# to these identifiers by various sources in the META network.
+
+#
+# --- Main Entities ---
+#
+type MusicPerformer {
+  identifiers: [PartyIdentifier]
+
+  name: StringValue
+
+  products:   [MusicProductLink]
+  releases:   [MusicReleaseLink]
+  recordings: [MusicRecordingLink]
+  labels:     [RecordLabelLink]
+}
+
+type MusicComposer {
+  identifiers: [PartyIdentifier]
+
+  name: StringValue
+
+  works:      [MusicWorkLink]
+  publishers: [MusicPublisherLink]
+}
+
+type RecordLabel {
+  identifiers: [PartyIdentifier]
+
+  name: StringValue
+
+  products:   [MusicProductLink]
+  releases:   [MusicReleaseLink]
+  recordings: [MusicRecordingLink]
+  performers: [MusicPerformerLink]
+}
+
+type MusicPublisher {
+  identifiers: [PartyIdentifier]
+
+  name: StringValue
+
+  composers: [MusicComposerLink]
+  works:     [MusicWorkLink]
+}
+
+type MusicRecording {
+  isrc: ID
+
+  title: StringValue
+
+  products:   [MusicProductLink]
+  releases:   [MusicReleaseLink]
+  performers: [MusicPerformerLink]
+  labels:     [RecordLabelLink]
+  works:      [MusicWorkLink]
+}
+
+type MusicWork {
+  iswc: ID
+
+  title: StringValue
+
+  composers:  [MusicComposerLink]
+  recordings: [MusicRecordingLink]
+  publishers: [MusicPublisherLink]
+}
+
+type MusicRelease {
+  grid: ID
+
+  title: StringValue
+
+  products:   [MusicProductLink]
+  recordings: [MusicRecordingLink]
+  performers: [MusicPerformerLink]
+  labels:     [RecordLabelLink]
+}
+
+type MusicProduct {
+  upc: ID
+
+  releases:   [MusicReleaseLink]
+  recordings: [MusicRecordingLink]
+  performers: [MusicPerformerLink]
+  labels:     [RecordLabelLink]
+}
+
+#
+# --- Link Types ---
+#
+type MusicPerformerLink {
+  source:    Source!
+  performer: MusicPerformer!
+  role:      String
+}
+
+type MusicComposerLink {
+  source:   Source!
+  composer: MusicComposer!
+}
+
+type RecordLabelLink {
+  source: Source!
+  label:  RecordLabel!
+}
+
+type MusicPublisherLink {
+  source:    Source!
+  publisher: MusicPublisher!
+}
+
+type MusicRecordingLink {
+  source:    Source!
+  recording: MusicRecording!
+}
+
+type MusicWorkLink {
+  source: Source!
+  work:   MusicWork!
+}
+
+type MusicReleaseLink {
+  source:  Source!
+  release: MusicRelease!
+}
+
+type MusicProductLink {
+  source:  Source!
+  product: MusicProduct!
+}
+
+#
+# --- Value Types ---
+#
+enum PartyIdentifierType {
+  ISNI
+  IPI
+  DPID
+}
+
+type PartyIdentifier {
+  type:   PartyIdentifierType!
+  value:  String!
+  source: Source!
+}
+
+type StringValue {
+  value:   String!
+  sources: [StringSource]
+}
+
+type StringSource {
+  value:  String!
+  source: Source!
+  score:  Int!
+}
+
+type Source {
```

Comment:
this Source type is here for clarity ? can it be other than string ?

Reply:
A source being just a string isn't very long term, it should at least be a META ID + name but we haven't got that far just yet.